src/main/java/com/yelp/nrtsearch/server/luceneserver/state/BackendGlobalState.java

```diff
@@ -177,6 +177,25 @@ StateBackend getStateBackend() {
     return stateBackend;
   }
 
+  @Override
+  public synchronized void reloadStateFromBackend() throws IOException {
+    GlobalStateInfo newGlobalStateInfo = getStateBackend().loadOrCreateGlobalState();
+    Map<String, IndexStateManager> newManagerMap = new HashMap<>();
+    for (Map.Entry<String, IndexGlobalState> entry :
+        newGlobalStateInfo.getIndicesMap().entrySet()) {
+      String indexName = entry.getKey();
+      IndexStateManager stateManager =
+          createIndexStateManager(indexName, entry.getValue().getId(), stateBackend);
```

Comment:
If the current `ImmutableState` has an `IndexStateManager` for this index and id, it should be reused instead of creating a new one.